AppController/djinn.rb

```diff
@@ -4744,74 +4747,75 @@ def check_running_apps()
           to_start << app
           no_appservers << app
         elsif not MonitInterface.is_running?("#{app}-#{port}")
-          Djinn.log_warn("Appserver for #{app} at port #{port} is not running.")
-          to_end << "#{app}:#{port}"
+          Djinn.log_debug("Didn't find the AppServer for #{app} at port #{port}.")
+          if @last_decision[app]
+            if Time.now.to_i - @last_decision[app] > APP_UPLOAD_TIMEOUT * RETRIES
+              Djinn.log_warn("AppServer for #{app} at port #{port} is not running.")
+              to_end << "#{app}:#{port}"
```

Comment:
good catch: there are 2 options, we just don't delete it here, or we define to_end before. I will check with you over IRC what you think is best.